condominium/data/ir_actions_server.xml

```diff
@@ -1,139 +1,362 @@
-        <field name="name">Distribute Costs</field>
-        <field name="model_id" ref="account.model_account_move"/>
-        <field name="state">code</field>
-        <field name="code"><![CDATA[
-x_ratios = record.sudo().x_distribution_key.x_ratios
-total_ratio = sum(x_ratios.mapped('x_ratio'))
-analytic_distribution = {}
-if total_ratio != 0:
-    for ratio in x_ratios.filtered(lambda l: l.x_ratio != 0):
-        analytic_distribution[ratio.x_property.id] = ratio.x_ratio / total_ratio * 100
-else: # If there is no pricelist/distribution key set, we use the area
-    properties = record.sudo().company_id.partner_id.x_condominium_properties
-    total_area = sum(properties.mapped('x_area'))
-    if total_area != 0:
-        for property in properties.filtered(lambda l: l.x_area != 0):
-            analytic_distribution[property.id] = property.x_area / total_area * 100
-for aml in record.invoice_line_ids:
-    aml['analytic_distribution'] = analytic_distribution
-]]></field>
-    </record>
     <record id="ir_act_server_confirm_action" model="ir.actions.server">
         <field name="name">Confirm</field>
         <field name="state">code</field>
         <field name="model_id" ref="sale.model_sale_order"/>
         <field name="binding_model_id" ref="sale.model_sale_order"/>
         <field name="code"><![CDATA[
 records.action_confirm()
-]]></field>
+]]>        </field>
     </record>
-    <record id="action_server_set_usage_meter_reading" model="ir.actions.server" >
-        <field name="name">Meter Reading</field>
-        <field name="model_id" ref="model_meter_reading"/>
+    <record id="confirm_analytic_account_item_split_action" model="ir.actions.server">
+        <field name="name">Confirm Account Analytic Line</field>
         <field name="state">code</field>
-        <field name="base_automation_id" ref="automation_set_usage_meter_reading"/>
+        <field name="model_id" ref="model_analytic_account_wizard"/>
+        <field name="binding_model_id" ref="model_analytic_account_wizard"/>
         <field name="code"><![CDATA[
-mrs = env['x_meter_reading'].search([
-    ('id', 'in', record.x_account_analytic_account_id.x_property_meter_reading_ids.ids),
-    ('x_meter_id', '=', record.x_meter_id.id)], order='x_date')
-previous_mr = False
-for mr in mrs:
-    mr['x_usage'] = mr['x_quantity'] - previous_mr['x_quantity'] if previous_mr else 0
-    previous_mr = mr
-]]>
+total_share = sum(line.x_share for line in records.x_line_ids)
+
+if round(total_share) != 100: raise UserError("The Share sum shall equal 100% to confirm the split.")
+
+new_analytic_item = env['account.analytic.line']
+note_details = {}
+occupied_cells = {}
+
+project_plan, other_plans = records.env['account.analytic.plan']._get_all_plans()
+for plan in other_plans:
+    if plan.name == 'Account':
+        fname = plan._column_name()
+
+for line in records.x_line_ids:
+    original_analytic = line.x_analytic_line_id or original_analytic
```

Comment:
original_analytic is used before assigned. If line.x_analytic_line_id is false, it will raise error.